src/treeland/quick/qml/WindowsSwitcher.qml

```diff
@@ -79,37 +86,144 @@ Item {
                 }
             }
         }
+        onCountChanged: if (visible) indicatorPlane.calcLayout()
     }
 
-    Rectangle {
-        width: switcher.width
-        height: switcher.height
-        anchors.centerIn: parent
-        radius: 10
-        opacity: 0.4
-    }
+    property int spacing: 10
+    property var rows: []
+    property int rowHeight: 0
+    property int padding: 8
+
+    Item {
+        id: indicatorPlane
+        
+        // currently use binding, so indicatorPlane follows mouse/active output
+        x: {
+            const coord=parent.mapFromItem(activeOutput, 0, 0)
+            return coord.x
+        }
+        y: {
+            const coord=parent.mapFromItem(activeOutput, 0, 0)
+            return coord.y
+        }
+        width: activeOutput?.width
+        height: activeOutput?.height
+        
+        MouseArea {
+            anchors.fill: parent
+            onClicked: {
+                root.visible = false
```

Comment:
ping

Reply:
will fix keyboard interaction in another PR